modules/n1ql/pages/n1ql-language-reference/upsert.adoc

```diff
@@ -31,31 +31,46 @@ If the statement has any RETURNING clauses, then the _Query Select_ privilege is
 For more details about user roles, see
 {authorization-overview}[Authorization].
 
+WARNING: Please note that the examples below will alter the data in your sample buckets.
+
 For example,
 
 * To execute the following statement, user must have the _Query Update_ and _Query Insert_ privileges on `pass:c[`travel-sample`]`.
 +
 [source,n1ql]
 ----
-UPSERT INTO `travel-sample` (KEY, VALUE)
-VALUES ("key1", { "type" : "hotel", "name" : "new hotel" })
+UPSERT INTO `travel-sample`.inventory.hotel (KEY, VALUE)
+VALUES ("key1", { "type" : "hotel", "name" : "new hotel" });
 ----
 
 * To execute the following statement, user must have the _Query Update_ and _Query Insert_ privileges on the `pass:c[`travel-sample`]` bucket and _Query Select_ privilege on `pass:c[`travel-sample`]` (for RETURNING clause).
 +
 [source,n1ql]
 ----
-UPSERT INTO `travel-sample` (KEY, VALUE)
+UPSERT INTO `travel-sample`.inventory.hotel (KEY, VALUE)
 VALUES ("key1", { "type" : "hotel", "name" : "new hotel" })
-RETURNING *
+RETURNING *;
+----
++
+.Result:
+[source,json]
+----
+[
+  {
+    "hotel": {
+      "name": "new hotel",
+      "type": "hotel"
+    }
+  }
+]
 ----
 
-* To execute the following statement, user must have the _Query Update_ and _Query Insert_ privileges on the `pass:c[`travel-sample`]` bucket and _Query Select_ privilege on `pass:c[`beer-sample`]`.
+* To execute the following statement, you must have the _Query Update_ and _Query Insert_ privileges on the `pass:c[`travel-sample`.inventory.hotel]` bucket and _Query Select_ privilege on `pass:c[`beer-sample`]`.
 +
 [source,n1ql]
 ----
-UPSERT INTO `travel-sample` (KEY foo, VALUE bar)
-SELECT foo, bar FROM `beer-sample`
```

Comment:
Similar issue here.

The error returned is:
```
 "Cannot UPSERT non-string key Missing field or index foo. of type value.missingValue."
```

Reply:
Yes this query wouldn't have worked previously. It's trying to insert records (key-value pairs) where the key is defined by the alias foo and the value is defined by the alias bar. In order for the insert to work, we have to set foo to the id of an existing record and set bar to the content of an existing record. See suggestion below.